studio/src/lib/schema-helpers.ts

```diff
@@ -650,9 +662,97 @@ export const getDeprecatedTypes = (
   return deprecatedTypes;
 };
 
+type ExtractedAuthenticationDirectives = {
+  authenticated: boolean;
+  requiresScopes: string[][] | null;
+}
+
+const extractAuthenticationDirectives = (
+    astNode: FieldDefinitionNode | TypeDefinitionNode | undefined | null
```

Comment:
Why do you expect undefined and null as potential inputs?

Reply:
The field `astNode` is optional, so instead of having a conditional check everywhere we expect to get the authentication directives, its just easier to have that check in a single place and exit early.

The definition for teh `astNode` field is `FieldDefinitionNode | undefined | null`

Reply:
Please see latest comment.